File: AULA_01/JAVA/operadores_precedencia.java
/* operadores_precedencia.java
 * Demonstração da precedência dos operadores em Java
 * Entrada: nenhum
 * Saída: nenhum
 *
 * Author: Fabrício Olivetti de França
 * Disciplina Processamento da Informação
 * Universidade Federal do ABC
 */

public class operadores_precedencia{
	public static void main(String[] args) {
		int a=20;
		int b=10;
		int c=15;
		int d=5;

		System.out.println( a+b*c-a+b*d ); // o que vai dar de resultado?
		System.out.println( a + (b*c) - a + (b*d) ); //Fica fácil ver a ordem de operações
		int e = (a + b) * c / d;
		System.out.println("Valor de (a + b) * c / d = " + e);

		e = ( (a+b) * c) / d;
		System.out.println("Valor de ((a + b) * c) / d = " + e);
		
		e = (a+b) * (c/d);
		System.out.println("Valor de (a + b) * (c / d) = " + e);

		e = a + (b * c) / d;
		System.out.println("Valor de a + (b * c) / d = " + e);

	}
}

File: AULA_01/C/operadores_precedencia.c
#include <stdio.h>

int main( ){

  int a,b,c,d;
  a = 20;
  b = 10;
  c = 15;
  d = 5;

  printf("%d\n", a+b*c-a+b*d ); // qual o resultado?
  printf("%d\n", a + (b*c) - a + (b*d) ); // mais claro
  int e;

  e = (a+b)*c/d;
  printf("Valor de (a+b)*c/d = %d\n",e);

  e = ((a+b)*c)/d;
  printf("Valor de ((a+b)*c)/d = %d\n",e);

  e = (a+b)*(c/d);
  printf("Valor de (a+b)*(c/d) = %d\n",e);

  e = a+(b*c/d);
  printf("Valor de a+(b*c/d) = %d\n",e);

  return 0;
}
